services/httpd/handler.go

```diff
@@ -517,25 +518,36 @@ func (h *Handler) serveWrite(w http.ResponseWriter, r *http.Request, user *meta.
 	}
 
 	h.statMap.Add(statPointsWrittenOK, int64(len(points)))
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
+
+	// Clients requesting gzipped response will receive StatusOK and
+	// minimal gzip response.
 }
 
 // serveOptions returns an empty response to comply with OPTIONS pre-flight requests
 func (h *Handler) serveOptions(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
 }
 
 // servePing returns a simple response to let the client know the server is running.
 func (h *Handler) servePing(w http.ResponseWriter, r *http.Request) {
 	h.statMap.Add(statPingRequest, 1)
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
 }
 
 // serveStatus has been depricated
 func (h *Handler) serveStatus(w http.ResponseWriter, r *http.Request) {
 	h.Logger.Printf("WARNING: /status has been depricated.  Use /ping instead.")
 	h.statMap.Add(statStatusRequest, 1)
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
```

Comment:
After thinking a little more, I'm pretty sure it should be the other way around – if the status is going to be a 204 and the response body is empty, then we should delete (or preferably never set) the `Content-Encoding` header.